File: 品味Java的21種設計模式/dp_cpp/command/example7/DuckCommand.cpp
#include "DuckCommand.h"

namespace cn {
namespace javass {
namespace dp {
namespace command {
namespace example7 {

void DuckCommand::setCookApi(CookApi *cookApi) {
    this->cookApi = cookApi;
}

DuckCommand::DuckCommand(int tableNum) {
    this->tableNum = tableNum;
}

int DuckCommand::getTableNum() {
    return this->tableNum;
}

void DuckCommand::execute() {
    this->cookApi->cook(tableNum, "北京烤鸭");
}
}
}
}
}
}

File: 品味Java的21種設計模式/dp_java/command/example7/DuckCommand.java
package cn.javass.dp.command.example7;

import java.io.Serializable;

/**
 * 命令对象，北京烤鸭
 */
public class DuckCommand implements Command, Serializable {
    private CookApi cookApi = null;
    public void setCookApi(CookApi cookApi) {
        this.cookApi = cookApi;
    }
    private int tableNum;
    public DuckCommand(int tableNum) {
        this.tableNum = tableNum;
    }
    public int getTableNum() {
        return this.tableNum;
    }

    public void execute() {
        this.cookApi.cook(tableNum, "北京烤鸭");
    }
}
